### ISSUE TITLE
Adding a new method to ivadomed, intervertebral disc labeling

### ISSUE THREAD

**Issue by rezazad68** [ivadomed/ivadomed#766]
Hi,

Would appreciate your comments and guidance regarding the process for adding a new method to the ivadomed library. I am following the below steps:

1- Add model to the [model.py](https://github.com/ivadomed/ivadomed/blob/a245bc4b1a4e2b8c34311d331b12cb93fa8de77b/ivadomed/models.py) 
2- Add loss function to [losses.py](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/losses.py)
3- Add metric to [metrics.py](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/metrics.py)
3- For training and evaluation, the [main.py](https://github.com/ivadomed/ivadomed/blob/a245bc4b1a4e2b8c34311d331b12cb93fa8de77b/ivadomed/main.py) should revise? 
4- weight files should be stored in which address? 
Any further comments are welcome. Thanks in advance.

**Comment by charleygros** [comment]
I agree with @andreanne-lemay !
The best way would probably be to start a pull request with your code and we'd be happy to guide you from there :-) Feel free to tag us wherever you have questions

**Comment by rezazad68** [comment]
The pose estimation method similar to the Countception approach uses the 2D images (mid slices) to train and evaluate the model. Thus, the config file uses the "T1w_mid". Since the ivadomed needs to be used by the end-user, I was wondering to know should I create a bash file to generate those samples (the user can easily run by address) or somehow include them in ivadomed?

**Comment by andreanne-lemay** [comment]
Ideally, we want one command to run the whole process (preprocessing, loading, training) and want to avoid multiple manual steps. This being said, I am not sure to understand what preprocessing step you are referring to. Lucas wrote this python script to prepare the dataset for vertebral labeling "ivadomed/scripts/prepare_dataset_vertebral_labeling.py". Are you referring to this? If so, you can start by simply reusing / adapting this script to the needs of your model.

**Comment by rezazad68** [comment]
Yes, I am referring to Lucas's script. This script needs to be run separately before running the ivadomed command, that is my point. If it is fine I can go in the same direction.

**Comment by andreanne-lemay** [comment]
Yes, this sounds like a good starting point!

**Comment by jcohenadad** [comment]
@rezazad68 indeed, this "preparation script" will ultimately be incorporated by the calling function, e.g., SCT's code. So you don't need to worry about incorporating these preprocessing steps into ivadomed. As long as you document the steps to generate a model, this is all fine.

**Comment by rezazad68** [comment]
The `ivadomed 2.7.4` source code doesn't have a `resize` transform, is there any specific reason for it?

**Comment by andreanne-lemay** [comment]
There is a resample transformation that allows resizing the image (https://github.com/ivadomed/ivadomed/blob/master/ivadomed/transforms.py#L211).

**Comment by jcohenadad** [comment]
Amazing! @rezazad68 could you please open a PR so we can comment on your changes.
